### DISCUSSION TITLE
Text to Speech help

### DISCUSSION THREAD

**Discussion by banyavaev** [snakers4/silero-models#258] [Q&A]
I run the following code
import time

import sounddevice as sd
import torch

language = 'ru'
model_id = 'v4_ru'
sample_rate = 48000  # 48000
speaker = 'aidar'  # aidar, baya, kseniya, xenia, random
put_accent = True
put_yo = True
device = torch.device('cpu')  # cpu или gpu
text = "проверка вывода звука"

model, _ = torch.hub.load(repo_or_dir='snakers4/silero-models',
                          model='silero_tts',
                          language=language,
                          speaker=model_id)
model.to(device)

def va_speak(what: str):
    audio = model.apply_tts(text=what + "..",
                            speaker=speaker,
                            sample_rate=sample_rate,
                            put_accent=put_accent,
                            put_yo=put_yo)

    sd.play(audio, sample_rate * 1.05)
    time.sleep((len(audio) / sample_rate) + 0.5)
    sd.stop()

and as a result I get: Using cache found in C:\Users\Vladimir/.cache\torch\hub\snakers4_silero-models_master
Please help me!

**Comment by snakers4** [comment]
What is the error?

**Reply by banyavaev** [reply]
the program had to convert the text into sound for me to hear it. But it says that the cache was found in the folder. Also, if instead of outputting sound I create an mp3 file with voiceover of the text, then everything works

**Reply by snakers4** [reply]
The cache message is a torch.hub message. You can load the model manually, if for some reason this is a problem.

**Comment by zoldaten** [comment]
@banyavaev perhaps u have error 'urllib.error.URLError: <urlopen error [WinError 10060]' ? there is a problem with torch.hub on windows.
try offline example:
```
import onnx
import torch
import onnxruntime
from omegaconf import OmegaConf
#https://huggingface.co/Derur/silero-models/tree/main/stt/ua/ua_v3

from utils import (init_jit_model, #utils from silero
                       split_into_batches,
                       read_batch,
                       prepare_model_input, Decoder)
#https://huggingface.co/Derur/silero-models/tree/main
#pip install torchaudio soundfile

# load provided utils
#_, decoder, utils = torch.hub.load(repo_or_dir='snakers4/silero-models', model='silero_stt', language=language, trust_repo=True)
#(read_batch, split_into_batches, read_audio, prepare_model_input) = utils
decoder='ua_v1_labels.json'

 # see available models
#torch.hub.download_url_to_file('https://raw.githubusercontent.com/snakers4/silero-models/master/models.yml', 'models.yml')
models = OmegaConf.load('models.yml')
available_languages = list(models.stt_models.keys())
print(available_languages)
#assert language in available_languages

# load the actual ONNX model
#torch.hub.download_url_to_file(models.stt_models.en.latest.onnx, 'model.onnx', progress=True)
onnx_model = onnx.load('ua_v3.onnx')
onnx.checker.check_model(onnx_model)
ort_session = onnxruntime.InferenceSession('ua_v3.onnx')


# download a single file, any format compatible with TorchAudio
#torch.hub.download_url_to_file('https://opus-codec.org/static/examples/samples/speech_orig.wav', dst ='speech_orig.wav', progress=True)
test_files = ['voice.wav']
batches = split_into_batches(test_files, batch_size=10)
input = prepare_model_input(read_batch(batches[0]))

# actual onnx inference and decoding
onnx_input = input.detach().cpu().numpy()
ort_inputs = {'input': onnx_input}
ort_outs = ort_session.run(None, ort_inputs)
#print(torch.Tensor(ort_outs[0])[0])
torch.set_grad_enabled(False)
model = torch.jit.load('ua_v3_jit.model', map_location='cpu')
model.eval()
#_, decoder = init_jit_model(model_url=models.stt_models.get(language).get(version).get('jit'),)
#https://github.com/snakers4/silero-models/blob/master/src/silero/utils.py#L50

decoder=Decoder(model.labels)
decoded = decoder(torch.Tensor(ort_outs[0])[0])
print(decoded)
```

use u lang, speakers and models.
